### PR TITLE
refactored email tests to get test coverage a bit higher

### PR DESCRIPTION
**Pull request by erikbern**

### CODE REVIEW

**Comment by Tarrasch** [comment]
Did we remove coveralls?

File: luigi/notifications.py [hunk 1]
```diff
@@ -113,6 +113,8 @@ def send_email_sendgrid(config, sender, subject, message, recipients, image_png)
 
 
 def send_email(subject, message, sender, recipients, image_png=None):
+    config = configuration.get_config()
```
Comment (Tarrasch) [inline luigi/notifications.py]: Looks like you're also changing logic and not only tests.

**Comment by Tarrasch** [comment]
LGTM, but please explain the non-test changes :)

**Comment by erikbern** [comment]
The non test changes were because we wouldn't send email id either (a) you
were running in debug mode (b) running in a terminal. I added a config that
lets you override that check and send email anyway. Otherwise you can't
test it

**Comment by erikbern** [comment]
Coveralls is still running but I disabled the comments on PR's since I thought they were too spammy

See here: https://coveralls.io/r/spotify/luigi

Coverage went up from about 77.5% to 78.3% with this PR. Nothing crazy. But the code was bothering me and hopefully the PR demonstrates how you can make everything testable if you want to,

File: luigi/notifications.py [hunk 2]
```diff
@@ -124,12 +126,10 @@ def send_email(subject, message, sender, recipients, image_png=None):
                  "-------------", recipients, sender, subject, message)
     if not recipients or recipients == (None,):
         return
-    if sys.stdout.isatty() or DEBUG:
+    if (sys.stdout.isatty() or DEBUG) and (not config.getboolean('email', 'force-send', False)):
```
Comment (themalkolm) [inline luigi/notifications.py]: Update docs too maybe?

Comment (erikbern) [inline luigi/notifications.py]: yeah i'm planning to change all config to use the new `Config` class anyway so I'll do it later

**Pull request merged**